app/server-utils/exchange-code-for-token.ts

```diff
@@ -0,0 +1,15 @@
+import { requireEnv } from './query-graphql';
+
+export async function exchangeCodeForToken(code: string) {
+  const clientId = requireEnv('GITHUB_CLIENT_ID');
+  const clientSecret = requireEnv('GITHUB_CLIENT_SECRET');
+  const URL = `https://github.com/login/oauth/access_token?client_id=${clientId}&client_secret=${clientSecret}&code=${code}`;
```

Comment:
For defense in depth, I would URL escape when splatting into query parameters. The vanilla JS way is (older) [`encodeURIComponent`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent) or (newer) [`URL.searchParams`](https://developer.mozilla.org/en-US/docs/Web/API/URL/searchParams) or [`URLSearchParams`](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams). Not sure if React or Next provides a wrapper API for that.